packages/augur-sdk/src/state/db/DB.ts

```diff
@@ -265,6 +397,55 @@ export class DB {
     await this.marketDatabase.sync(highestAvailableBlockNumber);
   }
 
+  async prune(timestamp: number) {
+    // Discover the markets that whose time has come.
+    const marketsToRemove = await this.MarketFinalized.where('timestamp')
+      .belowOrEqual(`0x${(timestamp - PRUNE_HORIZON).toString(16)}`)
+      .toArray();
+
+    if (marketsToRemove.length === 0) return;
+
+    const marketIdsToRemove = marketsToRemove.map(({ market }) => market);
+
+    // This should probably be calculated in the constructor.
+    const dbsToRemoveMarketsFrom = this.genericEventDBDescriptions
+      .filter(({ indexes, primaryKey }) =>
+        [...indexes, primaryKey].includes('market')
+      )
+      .map(({ EventName }) => EventName)
+      .filter((eventName) => eventName !== 'ShareTokenBalanceChanged');
+
+    await Promise.all(
+      [...dbsToRemoveMarketsFrom, 'Markets', 'MarketVolumeChangedRollup', 'MarketOIChangedRollup'].map(dbName =>
+        this[dbName]
+          .where('market')
+          .anyOf(marketIdsToRemove)
+          .delete()
+      )
+    );
+
+    // Now to deal with 'ShareTokenBalanceChanged' logs.
+    const rollupShareTokenBalanceChangedLogs = await this.ShareTokenBalanceChangedRollup.where(
```

Comment:
Why do we need to do special handling of the `ShareTokenBalanceChanged` table? Shouldn't just deleting anything with the matching market as normal be ok?

For the rollup this could also just be a delete on a `whereBetween` clause without querying any other table couldn't it?